scripts/globals/magic.lua

```diff
@@ -848,13 +865,20 @@ function addBonuses(caster, spell, target, dmg, params)
 
     dmg = math.floor(dmg * burst);
     local mabbonus = 0;
+    local spellId = spell:getID();
 
-    if (spell:getID() >= 245 and spell:getID() <= 248) then -- Drain/Aspir (II)
+    if (spellId >= 245 and spellId <= 248) then -- Drain/Aspir (II)
         mabbonus = 1 + caster:getMod(dsp.mod.ENH_DRAIN_ASPIR)/100;
-        -- print(mabbonus);
+        if (spellId == 247 or spellId == 248) and caster:getMainJob() == dsp.job.BLM and caster:getMainLvl() >= 75 then
+            mabbonus = mabbonus + caster:getMerit(dsp.merit.ASPIR_ABSORPTION_AMOUNT)/100
+        end
     else
         local mab = caster:getMod(dsp.mod.MATT) + params.bonusmab;
 
+        if spell:getSkillType() == dsp.skill.NINJUTSU and caster:getMainJob() == dsp.job.NIN and caster:getMainLvl() >= 75 then
```

Comment:
Remove main job and main level check. They are not needed. getMerit handles this.